src/main/java/dev/emi/trinkets/api/SlotAttributes.java

```diff
@@ -24,17 +24,20 @@ public static void addSlotModifier(Multimap<RegistryEntry<EntityAttribute>, Enti
 	}
 
 	public static Identifier getIdentifier(SlotReference ref) {
-		String key = ref.inventory().getSlotType().getId() + "/" + ref.index();
-		CACHED_IDS.computeIfAbsent(key, Identifier::of);
-		return CACHED_IDS.get(key);
+		String key = ref.getRefId();
+		return CACHED_IDS.computeIfAbsent(key, Identifier::of);
+	}
+
+	public static EntityAttributeModifier toSlotReferencedModifier(EntityAttributeModifier modifier, SlotReference ref){
+		return new EntityAttributeModifier(modifier.id().withSuffixedPath("/" + ref.getRefId()),modifier.value(),modifier.operation());
```

Comment:
Please put spaces between args. What is the use case of this method?

Reply:
Spaxes can be added. Explained in the original comment. Provides differentiation between modifiers made via nbt that otherwise all use the same nbt-provided identifier and collide when used in multiple places.

Reply:
I will note that I'm not very satisfied with the method location, but there is no `impl` package as I've noted elsewhere, so I'm a bit at a loss where I'd put methods like this that a user realistically never needs. I really don't feel like adding an entire package from my end. But maybe if combined with the other methods in `Trinket`, it would be a useful refactor to make.

Reply:
Doesn't nbt allow you to specify identifier per slot attribute? That is at least how it worked with uuids. It is sometimes desirable to have an accessory effect you cannot layer. Top level class `TrinketAttributes` with a handful of the static things we mentioned would be reasonable.

Reply:
Yes, you can define slots, but if you *don't* define a slot, the previous implementation also locks you to one stack of the effect max, which seems unintentional.

Reply:
Yes, but if you make this behavior the default you *cannot* make an effect with some uuid/id that is constant for all accessories, say, an accessory you can equip anywhere and gives you +10hp, but does not stack.

Reply:
Unsure how to resolve it. One or the other functionality is lost entirely without further capabilities of the Component.

Reply:
With the possibility of multiple slots per slot type, keeping the original "one-per-id" behavior seems less ideal. In the test mod for example, the "hat" adds 6 ring slots. So even if you define the ring slot in attribute NBT, you can only get one modifier despite the 7 identical slots; without some differentiation technique. 

In-code modifiers don't face this problem, they are given a unique id. based on slot reference id, which includes offset.

If only one slot was available per type, I could concede to vanilla behavior, which is one-per-id. But the behavior in-slot with multiple like-slots was one of my drivers to make the change in the first place, especially considering that it doesn't align to standard in-code behavior.